src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java

```diff
@@ -657,19 +658,21 @@
         BicycleOptimizeType optimize = BicycleOptimizeType.valueOf(environment.getArgument("optimize"));
 
         if (optimize == BicycleOptimizeType.TRIANGLE) {
-          callWith.argument("triangle.safetyFactor", request::setBikeTriangleSafetyFactor);
-          callWith.argument("triangle.slopeFactor", request::setBikeTriangleSlopeFactor);
-          callWith.argument("triangle.timeFactor", request::setBikeTriangleTimeFactor);
-          try {
-            RoutingRequest.assertTriangleParameters(
-                request.bikeTriangleSafetyFactor,
-                request.bikeTriangleTimeFactor,
-                request.bikeTriangleSlopeFactor
-            );
-          }
-          catch (ParameterException e) {
-            throw new RuntimeException(e);
-          }
+
+          // because we must use a final variable in the lambda we have to use this ugly crutch.
```

Comment:
We could add a new method `org.opentripplanner.ext.transmodelapi.support.DataFetcherDecorator`, where we could give a list/array of paths and a method which accepts that many arguments.

Reply:
I think adding a method to the `DataFetcherDecorator` with 3 args would be a bit hairy. So just using an array make it a bit more compact. 
```
          final double[] args = new double[3];

          callWith.argument("triangle.safetyFactor", (Double v) -> args[0] = v);
          callWith.argument("triangle.slopeFactor", (Double v) -> args[0] = v);
          callWith.argument("triangle.timeFactor", (Double v) -> args[0] = v);

          request.setTriangleNormalized(args[0], args[1], args[2]);
```
But, this is the Legacy API (not Transmodel API as I though in todays meeting), so the final call her is up to HSL I guess.

Reply:
Done.